pages/competition/[competitionId].js

```diff
@@ -1,18 +1,56 @@
 import { useRouter } from "next/router";
 import CompetitionDetail from "../../components/CompetitionDetail";
+import EditCompetition from "../../components/CompetitionEdit";
+import { useState } from "react";
+import { useContext } from "react";
```

Comment:
you can combine these two imports into 
```suggestion
import { useState, useContext } from "react";
```
Saves one line ;-)